addons/account/models/account_move.py

```diff
@@ -193,6 +193,13 @@ def _get_default_invoice_incoterm(self):
     country_code = fields.Char(related='company_id.account_fiscal_country_id.code', readonly=True)
     user_id = fields.Many2one(string='User', related='invoice_user_id',
         help='Technical field used to fit the generic behavior in mail templates.')
+    partner_shipping_id = fields.Many2one(
+        'res.partner',
+        string='Delivery Address',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]",
+        help="Delivery address for current invoice.")
```

Comment:
Let convert it from to a computed field!
```py
partner_shipping_id = fields.Many2one(
    comodel_name='res.partner',
    string="Delivery Address",
    domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]",
    compute='_compute_partner_shipping_id',
    store=True,
    readonly=False,
    help="Delivery address for current invoice.",
)
```

Since `states={'draft': [('readonly', False)]}` doesn't work with computed editable fields, you need to add manually `attrs={'readonly': [('state', '!=', 'draft')]}` on the view.

So, you don't need to do something specific on `_onchange_partner_id` and you could rewrite that as
```py
@api.depends('partner_id')
def _compute_partner_shipping_id(self):
    for move in self:
        if move.is_invoice(include_receipts=True):
            addr = move.partner_id.address_get(['delivery'])
            move.partner_shipping_id = addr and addr.get('delivery')
        else:
            move.partner_shipping_id = False
```

It seems you also can remove the `_get_invoice_delivery_partner_id` method because it's purpose was only to be overridden in `sale`.

Since `partner_shipping_id` depends of `partner_id`, you can directly adapt the fiscal position on `_onchange_partner_id` using `partner_shipping_id` and you can remove `_onchange_partner_shipping_id` I think

Reply:
This is just supposed to be a move of code, not some dev

Reply:
If we want to do the dev/fix in this PR, it needs to be in another commit

Reply:
@william-andre 
1) This is the perfect occasion to clean this part of the code
2) Don't say me you always do the strictly necessary diff on your task
3) mele-toi de tes fesses :)

It could be done in another commit, why not but imo, this is not strictly necessary.

Reply:
1. This is one way to end up with bugs, because this kind of tasks is generally not tested correctly because "it shouldn't change anything"
2. I often do more, it is true, but it is still something that I would do in a separate commit
3. Uh? Really? I reviewed this task and you came after I approved it. :mirror: I agree that I missed the invoice report, my bad.

Reply:
@william-andre We don't make pure refactoring task. So cleaning the code should be done when adding features/moving codes etc. The code is now OK but it doesn't cost much to convert this `onchange` to a computed field now.